roadmap/implementers-guide/src/types/overseer-protocol.md

```diff
@@ -33,6 +35,63 @@ struct ActiveLeavesUpdate {
 }
 ```
 
+## Approval Voting
+
+Messages received by the approval voting subsystem.
+
+```rust
+enum VoteCheckResult {
+	// The vote was accepted and should be propagated onwards.
+	Accepted,
+	// The vote was bad and should be ignored, reporting the peer who propagated it.
+	Bad,
+	// We do not have enough information to evaluate the vote. Ignore but don't report.
+	// This should occur primarily on startup.
+	Ignore,
+}
+
+enum ApprovalVotingMessage {
+	/// Check if the assignment is valid and can be accepted by our view of the protocol.
+	/// Should not be sent unless the block hash is known.
+	CheckAndImportAssignment(
+		Hash, 
+		AssignmentCert, 
```

Comment:
You've already checked the assignment cert here?  I'd expect the opposite

Reply:
We haven't checked it at this point, no.